### ISSUE TITLE
Set Up Screen Logical Error

### ISSUE THREAD

**Issue by Michaeldgeek** [systers/malaria-app-android#55]
I just noticed a logical error which I would try to describe.
If the user opens the app for the first time (fresh installation) and rotates his/her phone while at set up screen, the user is taken to the MainActivity without providing details.  This is caused by these lines of  code which i have corrected:
Logical error 

```
    @Override

   protected void onResume() {

    super.onResume();

    if (mSharedPreferenceStore.mPrefsStore.getBoolean("com.peacecorps.malaria.isFirstRun",
            true)) {  // false sould be passed instead of true

             mSharedPreferenceStore.mEditor.putBoolean(

                "com.peacecorps.malaria.hasUserSetPreference", true).commit();


    }

}

  @Override

protected void onPause() {

    super.onPause();
    if (mSharedPreferenceStore.mPrefsStore.getBoolean("com.peacecorps.malaria.isFirstRun",
            true)) { // false should be passed instead of true
        mSharedPreferenceStore.mEditor.putBoolean(
                "com.peacecorps.malaria.hasUserSetPreference", true).commit();
    }

}
```

Also to make sure that the user is not taken to set up screen, this line of code should be added in 

```
  saveUserTimeAndMedicationPrefs()
```

code:

```
    mSharedPreferenceStore.mEditor.putBoolean(
            "com.peacecorps.malaria.hasUserSetPreference", true).commit();
```

I have corrected them already. @ankita-kalra @rosariorobinson 
Should I go ahead and send a pull request?

**Comment by anubhakushwaha** [comment]
hey hi @Michaeldgeek I don't think the app has been designed for landscape mode so better leave it in the portrait mode as done by one of the students in his PR .
https://github.com/systers/malaria-app-android/pull/38

**Comment by yatna** [comment]
@Michaeldgeek I would like to give a explanation to this. Whenever the screen orientation is changed the activity lifecycle undergoes through the following:
onSaveInstanceState ->onPause ->onStop ->onCreate ->onStart ->onRestoreInstanceState ->onResume
Here this error occurs because checkInitialAppInstall() is called again on orientation change and this time it start the mainActivity. Also, the values selected by the user such as time picked, drug type etc. which were stored in variable and not yet saved in shared preferences are lost on changing screen orientation. 
A correct way to fix this would be to override the onSaveInstanceState() method like:
```java
public void onSaveInstanceState(Bundle savedInstanceState) {
super.onSaveInstanceState(savedInstanceState);
savedInstanceState.putInt("alarmHour", mHour);
savedInstanceState.putInt("alarmMinute", mMinute);
savedInstanceState.putInt("dayTakingDrug", checkDrug);
savedInstanceState.putString("drugPicked", mDrugPicked);
}
```
which you can later extract in the onRestoreInstanceState like
```java
@Override
public void onRestoreInstanceState(Bundle savedInstanceState) {
super.onRestoreInstanceState(savedInstanceState);
int mHour = savedInstanceState.getInt("alarmHour");
int myMinute = savedInstanceState.getInt("alarmMinute");
int checkDrug = savedInstanceState.getInt("dayTakingDrug");
String mDrugPicked = savedInstanceState.getString("drugPicked");
}
```
Thus the user setting would not be lost on changing the orientation.

**Comment by anubhakushwaha** [comment]
Closing,fixed via #133
